images/image.go

```diff
@@ -572,7 +573,30 @@ func cryptLayer(ctx context.Context, cs content.Store, desc ocispec.Descriptor,
 	}
 	// some operations, such as changing recipients, may not touch the layer at all
 	if resultReader != nil {
-		newDesc.Size, newDesc.Digest, err = content.WriteBlobUnchecked(ctx, cs, resultReader, content.WithExpiration(time.Minute*5))
+		cw, newDigest, size, err := content.WriteUnknownBlobUncommited(ctx, cs, resultReader)
+		if err != nil {
+			return ocispec.Descriptor{}, err
+		}
+		defer cw.Close()
+
+		if rsrcFn != nil {
+			r := leases.Resource{
+				ID:   newDigest.String(),
+				Type: "content",
+			}
+			if err := rsrcFn(r); err != nil {
+				return ocispec.Descriptor{}, errors.Wrap(err, "Unable to add resource to lease")
+			}
+		}
+
+		if err := cw.Commit(ctx, size, newDigest); err != nil {
+			if !errdefs.IsAlreadyExists(err) {
+				return ocispec.Descriptor{}, errors.Wrapf(err, "failed commit block")
+			}
+		}
+
+		newDesc.Digest = newDigest
+		newDesc.Size = size
```

Comment:
How about pushing all of this into a function:
```
newDesc.Size, newDesc.Digest, err = content.WriteBlobUncheckedLeased(ctx, cs, resultReader, rsrcFn)
```

Reply:
Yea agreed, let me see what a good way to factor this is, i will probably just pass the lease manager and lease around.